annotator/src/main/java/org/cbioportal/annotator/internal/GenomeNexusImpl.java

```diff
@@ -507,6 +510,25 @@ private String resolveConsequence() {
         }
         return "";
     }
+    
+    private String resolveEntrezGeneId(boolean replace) {
+        if (!replace) {
+            return mRecord.getEntrez_Gene_Id();
+        }
+        // make the rest call to ensembl server for gene external refs
+        RestTemplate restTemplate = new RestTemplate();
+        HttpEntity<LinkedMultiValueMap<String, Object>> requestEntity = getRequestEntity();        
+        ResponseEntity<GeneXref[]> responseEntity = restTemplate.exchange(geneXrefsServiceUrl + canonicalTranscript.getGeneId(), HttpMethod.GET, requestEntity, GeneXref[].class);
```

Comment:
I wonder about cases where a gene used to have multiple transcripts, but then the entrez ID was deprecated and split into two different genes ... one corresponding to one set of transcripts and one corresponding to a different set of transcripts.  In that case the two new genes may each have a canonicalTranscript.

When genomeNexus is called, is the transcript identified by the caller? Otherwise, can't it be difficult to choose between two such "split" genes?